hbase-common/src/main/java/org/apache/hadoop/hbase/io/crypto/tls/X509Util.java

```diff
@@ -234,11 +272,23 @@ public static SslContext createSslContextForClient(Configuration config)
 
     sslContextBuilder.enableOcsp(sslOcspEnabled);
     sslContextBuilder.protocols(getEnabledProtocols(config));
-    sslContextBuilder.ciphers(Arrays.asList(getCipherSuites(config)));
+    sslContextBuilder.ciphers(Arrays.asList(getCipherSuites(config, useOpenSsl)));
 
     return sslContextBuilder.build();
   }
 
+  private static boolean configureOpenSslIfAvailable(SslContextBuilder sslContextBuilder,
+    Configuration conf) {
+    if (OpenSsl.isAvailable() && conf.getBoolean(TLS_USE_OPENSSL, true)) {
+      LOG.debug("Using netty-tcnative to accelerate TLS handling");
+      sslContextBuilder.sslProvider(SslProvider.OPENSSL);
+      return true;
+    } else {
+      sslContextBuilder.sslProvider(SslProvider.JDK);
```

Comment:
The [default behavior](https://github.com/netty/netty/blob/4.1/handler/src/main/java/io/netty/handler/ssl/SslContext.java#L124-L130) is to use OPENSSL if its detected on the classpath. This is not safe for us because of the need to use `DEFAULT_CIPHERS_OPENSSL` when using openssl given our typical default ciphers. We could change our default ciphers overall, or simply `getDefaultCipherSuites` above to use just OpenSsl.isAvailable(). I prefer the config option as it gives users the ability to test and rollout tcnative if they desire, without needing to mess with the classpath.

Reply:
Might be useful to explicitly indicate in a warning message if user wants to use OpenSSL, but it's not available so HBase falls back to JDK. Netty is able to deal with this and log it if you just set the sslProvider to OPENSSL without making sure it's available.

ZooKeeper does the same btw: instead of relying on Netty's default, the ssl provider is always set explictly.